File: src/main/java/rs/raf/javaproject/controller/DelegateController.java
package rs.raf.javaproject.controller;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import rs.raf.javaproject.model.Point;
import rs.raf.javaproject.response.RegionStatusResponse;
import rs.raf.javaproject.response.ResultResponse;
import rs.raf.javaproject.service.JobService;

import java.util.Collection;

@RestController
@RequestMapping("api/delegate/{nodeID}")
    public class DelegateController {

        @Autowired
        JobService jobService;

    @GetMapping("jobs/{jobID}")
    @ResponseBody
    public Collection<Point> getJobResultFromNode(@PathVariable String nodeID, @PathVariable String jobID){
        return jobService.getJobResultFromNode(nodeID, jobID);
    }

    @GetMapping("/jobs/{jobID}/{regionID}")
    @ResponseBody
    public Collection<Point> getRegionResultFromNode(@PathVariable String nodeID, @PathVariable String jobID, @PathVariable String regionID){
        return jobService.getRegionResultFromNode(nodeID, jobID, regionID);
    }

    @GetMapping("/status/{jobID}")
    @ResponseBody
    public RegionStatusResponse myStatus(@PathVariable String nodeID, @PathVariable String jobID){
        return jobService.myStatus(nodeID, jobID);
    }
}

File: Controllers/DelegateController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace distributed_chaos_csharp.Services
{
    [Route("api/delegate")]
    [ApiController]
    public class DelegateController : ControllerBase
    {
        [HttpGet("/{nodeID}/jobs/{jobID}")]
        public IEnumerable<Point> getJobResultFromNode(@PathVariable String nodeID, @PathVariable String jobID)
        {
            return JobService.Instance.getJobResultFromNode(nodeID, jobID);
        }

        [HttpGet("/{nodeID}/jobs/{jobID}/{regionID}")]
        public IEnumerable<Point> getRegionResultFromNode(@PathVariable String nodeID, @PathVariable String jobID, @PathVariable String regionID)
        {
            return JobService.Instance.getRegionResultFromNode(nodeID, jobID, regionID);
        }

        [HttpGet("/{nodeID}/status/{jobID}")]
        public RegionStatusResponse myStatus(@PathVariable String nodeID, @PathVariable String jobID)
        {
            return JobService.Instance.myStatus(nodeID, jobID);
        }

        [HttpGet("/{nodeID}/status/{jobID}/{regionID}")]
        public RegionStatusResponse status(@PathVariable String nodeID, @PathVariable String jobID, @PathVariable String regionID)
        {
            return JobService.Instance.myStatus(nodeID, jobID, regionID);
        }

        [HttpGet("/{nodeID}/backup/{jobID}/{regionID}")]
        public BackupInfo getBackup(@PathVariable String nodeID, @PathVariable String jobID, @PathVariable String regionID)
        {
            return JobService.Instance.getBackup(nodeID, jobID, regionID);
        }
    }
}
